### PR TITLE
Bank reporter

### PR DESCRIPTION
**Pull request by KrasimirRaikov**
Handling exceptions with a filter(HttpReporterFilter) printing messages to the ErrorPage.html
regex validating whole numbers, validating the withdraw for balance having enough money
extracted the reading of the html

### CODE REVIEW

File: src/main/java/com/clouway/bank/http/HttpReporterFilter.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+
+/**
+ * Catching exceptions and printing messages to an Error page
+ *
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class HttpReporterFilter implements Filter {
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/bank/http/HttpReporterFilter.java]: rename to ```HttprequestErrorReporter```. Altrenative way to make this reporter is use: 
http://aopalliance.sourceforge.net/doc/org/aopalliance/intercept/MethodInterceptor.html.. 

Your Reporter should Implements this interface. When you lern about Guice in next tasks you can use method ```bindInterceptor``` of ```com.google.inject.AbstractModule```  to bind your HttpReportere.

I know that not you do not understand my but after some weeks and when read for Guice you will understand.

File: src/main/java/com/clouway/bank/core/FatalException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.clouway.bank.core;
+
+/**
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class FatalException extends RuntimeException {
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/bank/core/FatalException.java]: FatalException is not good name. Maybe UserException("No such user"), or TransactactionException("We not have enough many to make this transaction")

File: src/main/java/com/clouway/bank/http/HttpReporterFilter.java (hunk 1 above)
Comment (KrasimirRaikov) [inline src/main/java/com/clouway/bank/http/HttpReporterFilter.java]: renamed.

File: src/main/java/com/clouway/bank/core/FatalException.java (hunk 2 above)
Comment (KrasimirRaikov) [inline src/main/java/com/clouway/bank/core/FatalException.java]: renamed to TransactionException

File: src/main/java/com/clouway/utility/Reader.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.clouway.utility;
+
+/**
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public interface Reader {
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/utility/Reader.java]: Domumentation

Comment (KrasimirRaikov) [inline src/main/java/com/clouway/utility/Reader.java]: Done.

File: src/main/java/com/clouway/utility/Reader.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.clouway.utility;
+
+/**
+ * Reading the contents of a file
+ *
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/utility/Reader.java]: Name of interface is common ```Reader``` but method name is concrete ```getFile````. This is smell if name of interface are common its name also should be common.

Comment (KrasimirRaikov) [inline src/main/java/com/clouway/utility/Reader.java]: I've renamed the method to ```read(String source)```

File: src/main/java/com/clouway/utility/Template.java [hunk 5]
```diff
@@ -16,6 +16,12 @@
   void setTemplate(String template);
 
   /**
+   * uses reader to get a template
+   * @param url
+   */
+  void templateFromFile(String url);
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/utility/Template.java]: duplication of naming template is not good ```Template.templateFromFile```, maybe ```Template.loadFromFile(...)```

Comment (KrasimirRaikov) [inline src/main/java/com/clouway/utility/Template.java]: Done.

File: src/main/java/com/clouway/bank/persistent/PersistentAccountRepository.java [hunk 6]
```diff
@@ -76,7 +81,7 @@ public Double withdraw(Amount amount) throws ValidationException {
         throw new ValidationException(validationMessage);
       }
     } catch (SQLException e) {
-      throw new ValidationException("no such user");
+      throw new TransactionException("Sorry, can't find user with this name");
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/bank/persistent/PersistentAccountRepository.java]: here is more appropriate if you have another exception ```UserException``` and use it where have problem with user

Comment (KrasimirRaikov) [inline src/main/java/com/clouway/bank/persistent/PersistentAccountRepository.java]: Done.

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request merged**